src/airflow_tools/providers/filesystem/operators/filesystem.py

```diff
@@ -1,17 +1,75 @@
-from typing import Optional, Protocol
+from typing import TYPE_CHECKING, Optional, Protocol
 
 from airflow.hooks.base import BaseHook
 from airflow.models import BaseOperator
 from airflow.providers.common.sql.hooks.sql import DbApiHook
 
 from airflow_tools.filesystems.filesystem_factory import FilesystemFactory
 
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+import logging
+
+logger = logging.getLogger(__file__)
+
 
 class Transformation(Protocol):
     def __call__(self, data: bytes, filename: str, context: dict) -> bytes:
         ...
 
 
+class SQLToFilesystem(BaseOperator):
+    """
+    Copies data from a SQL query to a filesystem.
+    """
+
+    template_fields = (
+        'sql',
+        'destination_path',
+    )
+
+    def __init__(
+        self,
+        source_sql_conn_id: str,
+        destination_fs_conn_id: str,
+        sql: str,
+        destination_path: str,
+        batch_size: Optional[int] = 100000,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.source_sql_conn_id = source_sql_conn_id
+        self.destination_fs_conn_id = destination_fs_conn_id
+        self.sql = sql
+        self.destination_path = destination_path
+        self.batch_size = batch_size
+        self.files = []
+
+    def execute(self, context):
+        source_sql_hook: DbApiHook = BaseHook.get_connection(
+            self.source_sql_conn_id
+        ).get_hook()
+
+        destination_fs_hook = FilesystemFactory.get_data_lake_filesystem(
+            connection=BaseHook.get_connection(self.destination_fs_conn_id),
+        )
+
+        self.files.clear()
+        for i, df in enumerate(
```

Comment:
Does this work when `batch_size` is None or do we need to add a special case to fetch all records?

Reply:
It should. DbApiHook.get_pandas_df_by_chunks uses the function pandas.io.sql.read_sql which, for the parameter chunksize accepts either an int or None. 

[https://github.com/pandas-dev/pandas/blob/main/pandas/io/sql.py](https://github.com/pandas-dev/pandas/blob/main/pandas/io/sql.py) on line 528